### PR TITLE
WIXBUG:4929 - Fix infinite loop in PathCreateTimeBasedTempFile

### PR DESCRIPTION
**Pull request by rseanhall**
I also included writing an error entry to the Application event log if unable to create the Failed log. Feel free to skip that commit if it's not right for v3.

Fixes wixtoolset/issues#4929

### CODE REVIEW

File: src/burn/engine/logging.cpp [hunk 1]
```diff
@@ -158,6 +159,33 @@ extern "C" HRESULT LoggingOpen(
     return hr;
 }
 
+extern "C" void LoggingOpenFailed()
+{
+    HRESULT hr = S_OK;
+    HANDLE hEventLog = NULL;
+    LPCWSTR* lpStrings = (LPCWSTR*)&LOG_FAILED_EVENT_LOG_MESSAGE;
```
Comment (barnson) [inline src/burn/engine/logging.cpp]: const_cast is preferred.

File: src/libs/dutil/pathutil.cpp [hunk 2]
```diff
@@ -678,8 +680,11 @@ DAPI_(HRESULT) PathCreateTimeBasedTempFile(
             {
                 ::Sleep(100);
 
-                er = ERROR_SUCCESS;
-                fRetry = TRUE;
+                if (dwAttempts < 10)
```
Comment (barnson) [inline src/libs/dutil/pathutil.cpp]: Yoda-ize.

**Comment by rseanhall** [comment]
Done

**Comment by barnson** [comment]
:shipit:

**Pull request merged**